frontend/src/components/CollectibleCard.tsx

```diff
@@ -3,8 +3,24 @@ import { fetchJSONfromURI, ipfsToHttps } from "@/lib/utils";
 import Image from "next/image";
 import Link from "next/link";
 
-export default function CollectibleCard({ item, tokenURI, linkToOS }) {
-  let [metadata, setMetadata] = useState({});
+interface Metadata {
+  name?: string;
+  image?: string;
+  tokenId?: string;
+}
+
+interface CollectibleCardProps {
```

Comment:
@ZYJLiu not a problem for now, but curious if there's a convention moving forward that makes it so not all properties are optional. For example, the collectible card doesn't work without some of these fields so those fields should not be optional. We should find a way to reflect that at some point.